woocommerce/woocommerce-ios
# [Mobile Payments] Pass statement descriptor when initiating a payment capture

Closes #3938 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️

## Changes
* Remove the previously hardcoded receiptDescription and statementDesxription
* Pass a generic string containing the store name as [receiptDescription](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)stripeDescription)
* Pass WCPayAccount.statementDescriptor as [statementDescription](https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPPaymentIntentParameters.html#/c:objc(cs)SCPPaymentIntentParameters(py)statementDescriptor) This is the one we really care about as it is the one that might be in the customer's credit card statement.

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. The pre-release version of WCPay that implements the endpoint that captures the payment id (see p91TBi-54R#comment-4441-p2 for a downloadable zip)

* Checkout the branch, build and run
* Add a breakpoint to StripeCardReaderService, line 163
<img width="350" alt="Screen Shot 2021-05-11 at 06 11 19" src="https://user-images.githubusercontent.com/2722505/117832582-9b1f3a00-b243-11eb-926b-a6685409e758.png">
* Navigate to an eligible order, tap "Collect Payment"
* The breakpoint should be hit. `po parameters` should print something similar to this:

```
PaymentIntentParameters(amount: 200, _currency: Hardware.CurrencyCode(value: "USD"), receiptDescription: Optional("Receipt from wooctarda”), _statementDescription: Hardware.StatementDescriptor(value: Optional("CTARDAWOO.WPCOMSTAGING")), _receiptEmail: Hardware.EmailSwift.String(value: Optional("ctarda@automattic.com")), metadata: Optional([AnyHashable("paymentintent.storename"): Optional("wooctarda")]))
```

Notice receiptDescription and statementDescription. Actual values would depend on the store configuration, but data should be there. 

Update release notes:
